src/lit-html.ts

```diff
@@ -223,9 +220,9 @@ export class Template {
         }
       } else if (node.nodeType === 3 /* Node.TEXT_NODE */) {
         const nodeValue = node.nodeValue!;
-        const strings = nodeValue.split(attrOrTextRegex);
-        if (strings.length > 1) {
+        if (nodeValue.indexOf(marker) > -1) {
```

Comment:
Have I mentioned how insanely fast `String#indexOf` is?

This really optimizes for the negative cases (`Text` nodes without any markers in them) which is likely much more common. Splitting with a regex (which is kinda slow) can come after we know for sure we have a hit.